### DISCUSSION TITLE
Element type with cubic Lagrange interpolation

### DISCUSSION THREAD

**Discussion by ZhouXD1989** [idaholab/moose#16198] [Q&A General] [answered]
Hello all,

I found higher order element using cubic Lagrange interpolation in the following website:
https://mooseframework.inl.gov/magpie/source/problems/FEProblemBase.html  

Does moose provide this type of element for 2D and 3D problems?

Best wishes,
Zhou

**Comment by dschwen** [comment]
Given that your link is identical to https://mooseframework.inl.gov/source/problems/FEProblemBase.html (the MOOSE documentation) I think we can confidently answer this with "yes".

**Reply by dschwen** [reply]
too confident

**Comment by ZhouXD1989** [comment, marked as answer]
Dear Daniel,

Thanks for your reply.
Does moose have an introduction of the element library?
I already found moose have QUAD9 and HEX27 with quadratic basis function,
but what is the element name with cubic basis function?
I am working with gradient elasticity recently, that's why I want to try
higher order elements in moose. Does moose have other type of higher order
elements for gradient elasticity?

Thank you for your time and effort.

Best wishes,
Xiandong

Daniel Schwen <notifications@github.com> 于2020年11月17日周二 下午4:10写道：

> Given that your link is identical to
> https://mooseframework.inl.gov/source/problems/FEProblemBase.html (the
> MOOSE documentation) I think we can confidently answer this with "yes".
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/16198#discussioncomment-131685>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALDXGIBAFD5ADPBCTEDNURDSQKG47ANCNFSM4TW6T43Q>
> .
>


-- 
Best regards,
Xian-Dong Zhou
Mechanics of Functional Materials, Institution of Materials Science, TU
Darmstadt
E-mail: zhouxd22@g <taoyong@pku.edu.cn>mail.com

**Reply by dschwen** [reply]
Hey, ugh, yes. I just realized we have second order elements, but not third order elements. So you might have to settle for a hermite shape function!

**Reply by jiangwen84** [reply]
Daniel is correct, and Hermite shape function is third order shape function with C1 continuity. However, Hermite shape function does not work with the displaced mesh. So only small strain problem would work.
